source/plugin/internals/mcp-setup.rst

```diff
@@ -0,0 +1,114 @@
+====================
+Using MCP in Plugins
+====================
+
+ForgeGradle_ is a Gradle plugin which integrates the MCP workflow into the Gradle build system. It handles setting up
+the workspace, as well as the re-obfuscation of your plugin.
+
+.. note::
+    Since ForgeGradle depends on Gradle, the following pages assume you are using Gradle to build your plugin. See
+    :doc:`../project/gradle` to get started.
+
+Configuring ForgeGradle
+-----------------------
+You can choose between two different types of workspaces:
+
+- **Vanilla workspace:** Supports plugins for SpongeVanilla **and** SpongeForge.
+- **Forge workspace:** Supports **only** plugins for SpongeForge (and **not** SpongeVanilla).
+
+.. note::
+    In most cases, the Vanilla workspace can be used for SpongeVanilla and SpongeForge. In some cases, there may be
+    problems on one of the platforms because of changes in the Minecraft code by Forge. Make sure to always test your
+    plugin on both platforms when using MCP.
+
+Choosing a MCP mappings version
+```````````````````````````````
+To setup a MCP workspace you need to specify the MCP mappings version that will be used to de-obfuscate the Minecraft
+source with human-readable names. A list of MCP mappings versions is available on the
+`Export page of the MCPBot <http://export.mcpbot.bspk.rs>`_.
+
+There are stable versions (released from time to time) and daily snapshots which contain the latest name changes. If you
+do not need a specific name that was added recently, use a stable version (if available for your Minecraft version),
+otherwise use the latest snapshot version.
+
+Click the button for the version you want to use and select "Use in ForgeGradle". Then copy the provided version to your
+Gradle build script (insert it in the ``snapshot_xxx`` placeholder below).
+
+Vanilla Workspace
+`````````````````
+.. code-block:: groovy
+
+    buildscript {
+        repositories {
+            maven {
+                name = 'forge'
+                url = 'http://files.minecraftforge.net/maven'
+            }
+        }
+
+        dependencies {
+            classpath 'net.minecraftforge.gradle:ForgeGradle:2.2-SNAPSHOT'
+        }
+    }
+
+    plugins {
+        id 'org.spongepowered.plugin' version '0.7'
+        id 'net.minecrell.vanillagradle.server' version '2.2-3'
+    }
+
+    minecraft {
+        version = '1.10.2'
+        mappings = 'snapshot_xxx' // TODO
```

Comment:
You should mention that here instead of "TODO", then.
`mappings = 'YOUR_MAPPINGS_VERSION' // replace with your mappings version, eg: snapshot_20170120`
or something